maestro-domain/src/main/java/bio/overture/maestro/domain/api/DefaultIndexer.java

```diff
@@ -65,24 +68,39 @@
     private static final String CONFLICTS = "conflicts";
 
     private final FileCentricIndexAdapter fileCentricIndexAdapter;
+    private final AnalysisCentricIndexAdapter analysisCentricIndexAdapter;
     private final StudyDAO studyDAO;
     private final StudyRepositoryDAO studyRepositoryDao;
     private final ExclusionRulesDAO exclusionRulesDAO;
     private final Notifier notifier;
 
     @Inject
     DefaultIndexer(FileCentricIndexAdapter fileCentricIndexAdapter,
+                   AnalysisCentricIndexAdapter analysisCentricIndexAdapter,
                    StudyDAO studyDAO,
                    StudyRepositoryDAO studyRepositoryDao,
                    ExclusionRulesDAO exclusionRulesDAO,
                    Notifier notifier) {
         this.fileCentricIndexAdapter = fileCentricIndexAdapter;
+        this.analysisCentricIndexAdapter = analysisCentricIndexAdapter;
         this.studyDAO = studyDAO;
         this.studyRepositoryDao = studyRepositoryDao;
         this.exclusionRulesDAO = exclusionRulesDAO;
         this.notifier = notifier;
     }
 
+    @Override
+    public Mono<IndexResult> indexAnalysisToAnalysisCentric(@NonNull IndexAnalysisCommand indexAnalysisCommand){
```

Comment:
so right now this is only called through the http api right? and only for one analysis
you would also need it to be called as result of kafka event and as result of indexing a full study/repository

we can have a call if you need more info here

Reply:
I have another tickets for indexing a study and repo, this pr is just for indexing analysis, I wanted to break the task down.. But I'm not sure what you mean by calling the endpoint as a result of kafka event.